docs/src/main/asciidoc/writing-native-applications-tips.adoc

```diff
@@ -344,6 +344,231 @@ com.oracle.svm.core.jdk.UnsupportedFeatureError: Proxy class defined by interfac
 Solving this issue requires adding the `-H:DynamicProxyConfigurationResources=<comma-separated-config-resources>` option and to provide a dynamic proxy configuration file.
 You can find all the information about the format of this file in https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/DynamicProxy.md#manual-configuration[the GraalVM documentation].
 
+[[modularity-benefits]]
+=== Modularity Benefits
+
+As already explained in sections above,
+during native image build time GraalVM analyses the application's call tree and generates a universe of all the code it needs.
+Having a modular codebase is key to avoiding problems with unused or optional parts of your application,
+while at the same time reducing both native image build times and native executable size.
+In this section you will learn about the details behind the benefits of modularity for native applications.
+
+When code is not modular enough, generated native executables can end up with more code than what the user needs.
+If a feature is not used and the code gets compiled into the native executable,
+this is both a waste of native compilation time and native executable disk space.
```

Comment:
Isn't it also a waste of memory at the Quarkus level, given we initialize most classes at build time?